Mathlib/Analysis/Calculus/MeanValue.lean

```diff
@@ -590,11 +590,63 @@ theorem _root_.is_const_of_fderiv_eq_zero
 one point in that set, then they are equal on that set. -/
 theorem eqOn_of_fderivWithin_eq (hs : Convex ℝ s) (hf : DifferentiableOn 𝕜 f s)
     (hg : DifferentiableOn 𝕜 g s) (hs' : UniqueDiffOn 𝕜 s)
-    (hf' : ∀ x ∈ s, fderivWithin 𝕜 f s x = fderivWithin 𝕜 g s x) (hx : x ∈ s) (hfgx : f x = g x) :
+    (hf' : s.EqOn (fderivWithin 𝕜 f s) (fderivWithin 𝕜 g s)) (hx : x ∈ s) (hfgx : f x = g x) :
     s.EqOn f g := fun y hy => by
   suffices f x - g x = f y - g y by rwa [hfgx, sub_self, eq_comm, sub_eq_zero] at this
   refine hs.is_const_of_fderivWithin_eq_zero (hf.sub hg) (fun z hz => ?_) hx hy
-  rw [fderivWithin_sub (hs' _ hz) (hf _ hz) (hg _ hz), sub_eq_zero, hf' _ hz]
+  rw [fderivWithin_sub (hs' _ hz) (hf _ hz) (hg _ hz), sub_eq_zero, hf' hz]
+
+/-- If `f` has zero derivative on an open set, then `f` is locally constant on `s`. -/
```

Comment:
Since you say "locally constant" here, it would be nice to link this (or a version of this) to https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/LocallyConstant/Basic.html#IsLocallyConstant

Reply:
Unfortunately we don't have "IslocallyConstantOn". Do you want "Islocallyconstant (f \circ Subtype.val)" or a version specialized to Set.univ?

Reply:
Oof, fair point. How about a version specialized to Set.univ and a comment logging that IsLocallyConstantOn could be worth adding